tests/python/unittest/test_tvmscript_syntax_sugar.py

```diff
@@ -0,0 +1,72 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=missing-function-docstring,missing-module-docstring,invalid-name,pointless-string-statement
+import sys
+
+import pytest
+import ast as py_ast
+import tvm
+from tvm.script import tir as T
+
+# match buffer - no syntax sugar
+@T.prim_func
+def elementwise_handle(
+    a: T.handle,
+    b: T.handle,
+) -> None:
+    A = T.match_buffer(a, (128, 128, 128, 128))
+    B = T.match_buffer(b, (128, 128, 128, 128))
+    for i, j, k, l in T.grid(128, 128, 128, 128):
+        with T.block("B"):
+            vi, vj, vk, vl = T.axis.remap("SSSS", [i, j, k, l])
+            B[vi, vj, vk, vl] = A[vi, vj, vk, vl] * 2.0
+
+
+# match buffer - use buffer with kwargs
+@T.prim_func
+def elementwise_buffer_kwargs(
+    a: T.Buffer(shape=(128, 128, 128, 128), dtype="float32", name="a", elem_offset=None),
+    b: T.Buffer(shape=(128, 128, 128, 128), dtype="float32", name="b", elem_offset=None),
+) -> None:
+    for i, j, k, l in T.grid(128, 128, 128, 128):
+        with T.block("B"):
+            vi, vj, vk, vl = T.axis.remap("SSSS", [i, j, k, l])
+            b[vi, vj, vk, vl] = a[vi, vj, vk, vl] * 2.0
+
+
+# match buffer - use buffer without kwargs
+# This function is commented out as it is supported yet
+@T.prim_func
+def elementwise_buffer_no_kwargs(
+    a: T.Buffer[(128, 128, 128, 128), "float32", "a"],
```

Comment:
What's the difference between `Buffer[]` and `Buffer()`?

Reply:
They are two ways of utilizing the match buffer syntax sugar. `Buffer[]` is for unnamed positional arguments and `Buffer()` is for named keywords arguments.

Reply:
what happened if users don't specify buffer name `a` here, e.g. a: T.Buffer[(128, 128, 128, 128), "float32"]

Reply:
Those two buffers will have the default name "Buffer" and "Buffer_1"

Reply:
@Hzfengsy can you take a look of the latest change and approve so we can merge this PR in? Thanks!

Reply:
I will second Wuwei on the buffer name. Let's fix it before merging this PR

Reply:
I suggest the buffer name `a` in case `a: T.Buffer[(128, 128, 128, 128), "float32"]`. Also, we'd better use name `a_handle` for the func param

Reply:
this is addressed in the latest change